e2e/utils/src/installPackages.ts

```diff
@@ -0,0 +1,50 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as logSymbols from 'log-symbols';
+
+import { sh } from './sh';
+
+export async function installPackages(options: {
+  packages: string[];
+  resolutions: { file?: string; version?: string; packageName: string }[];
+  rootDir: string;
+  tempDir: string;
+}) {
+  const { tempDir, packages, resolutions, rootDir } = options;
+
+  const packageJsonPath = path.resolve(tempDir + '/package.json');
+  const packageJsonPathExists = !!(await fs.promises.stat(packageJsonPath).catch(() => false));
+
+  if (!packageJsonPathExists) {
+    throw new Error(`A "package.json" in a temporary directory does not exist`);
+  }
+
+  let packagesWithVersions = {};
+
+  if (packages.length > 0) {
+    const yarnOutput = await sh(`yarn info ${packages.join(' ')} --json`, rootDir, true);
+    const parsedYarnOutput = yarnOutput
+      .split('\n')
+      .filter(Boolean)
+      .map(line => JSON.parse(line));
+
+    packagesWithVersions = Object.fromEntries(
+      parsedYarnOutput.map(info => [info.value.split('@patch')[0].split('@npm')[0], info.children.Version]),
```

Comment:
what kind of package would have `@patch`?

Reply:
![image](https://user-images.githubusercontent.com/14183168/203842278-8164a542-7506-4e39-8fec-2392ff0a26b0.png)

Yarn applies a patch to TS.

Reply:
I will probably cover some of these utils with tests in upcoming PRs.